### ISSUE TITLE
Update SDK to 3.0.1

### ISSUE THREAD

**Issue by TheChatty** [jeelabs/esp-link#469]
Please update SDK to [v3.0.1](https://github.com/espressif/ESP8266_NONOS_SDK/archive/v3.0.1.zip).

**Comment by uzi18** [comment]
why?

**Comment by TheChatty** [comment]
Because I still fail to run any version above 2.2.3. On my ESP-01S (8mbit) those versions reboot a few seconds after boot repeatedly. Maybe it's due some changed default setting in esp-link, which causes the immediate reboot. But as v2.2.3 does not support OTA upgrade I can't upgrade with my settings in place, can I? I just hope current SDK somehow solves my reboot problem.
Where can I download the auto builds, esp. the latest master build?

Side note: [nodemcu](https://nodemcu.readthedocs.io/en/master/flash/#determine-flash-size) states: "_The SDK reserves space in flash that is used to store calibration and other data. This data changes between SDK versions, and if it is **invalid or not present**, the firmware may not boot correctly. Symptoms include messages like rf_cal[0] !=0x05,is 0xFF, or **endless reboot loops and/or fast blinking module LEDs**._" -- Exactly what I experience. And they say: "_NodeMCU will take care of writing the SDK init data when needed._ "

**Comment by TheChatty** [comment]
Accidentally I found a workaround. After flashing and running 2.2.3 and changing a few settings (esp-01 defaults, different baud rate etc.) I dirty flashed (no erasing) 3.2.47 (all four files: boot, user1, esp_init, blank) and it's working stable. I could even go ahead and flash my own compiled 09474e6 and it's working, too.

**Comment by uzi18** [comment]
can you find a way to reproduce this problem?
provide flash dump with it
I can try to investigate it further

**Comment by TheChatty** [comment]
Ok... I started with clean flash of v2.2.3:
```
python -m esptool --port com6 --baud 115200 write_flash --erase-all -fm dio ^
    0x00000 boot_v1.7.bin ^
    0x01000 user1_2.2.3.bin
```
I booted it and changed defaults to esp-01. I then dirty flashed v.3.2.47+ (meaning 09474e6):
```
python -m esptool --port com6 --baud 115200 write_flash -fm dio ^
    0x00000 boot_v1.7.bin ^
    0x01000 user1_3.2.47-09474e6.bin ^
    0xFC000 esp_init_data_default_v08.bin ^
    0xFE000 blank.bin
```
This results in a stable boot of 3.2.47. 

[dump_3.2.47+_after_223_set_to_esp01_working.bin.zip](https://github.com/jeelabs/esp-link/files/3797991/dump_3.2.47%2B_after_223_set_to_esp01_web_still_claims_223_console_says_3247_working.bin.zip)

But when I was trying today I also managed to get a different image where I thought I did the same as above but it is rebooting endlessly:
[dump_3.2.47+_after_223_set_to_esp01_stillAP_NOT_working.bin.zip](https://github.com/jeelabs/esp-link/files/3798103/dump_3.2.47%2B_after_223_set_to_esp01_stillAP_NOT_working.bin.zip)
I hope the mostly matching parts (only few differences) will help you in determining the root cause.

**Comment by TheChatty** [comment]
@uzi18: Are you investigating this?

**Comment by uzi18** [comment]
yes, but first need to manage a little bit of spare time for this, thanks for dumps will try to test

**Comment by TheChatty** [comment]
I think I found out the root cause. I was always using [this adapter](https://de.aliexpress.com/item/32806315523.html).

I now changed my setup to do without that adapter and the esp is much more stable, I can use higher baud rates than 115200 for flashing and so on. And when I tried a clean flash of a current esp-link release (--erase all and flash all four bins) it just works as expected.

Since getting rid of that stupid adapter I even successfully connected my heating regulator over WiFi.
